File: app/src/main/java/com/yamamz/santosrosaryo/MainActivity.java
package com.yamamz.santosrosaryo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.yamamz.santosrosaryo.adapter.mysteryAdapter;
import com.yamamz.santosrosaryo.model.Mystery;
import com.yamamz.santosrosaryo.ui.DeviderItemDecoration;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private List<Mystery> mysteries=new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        setupRecyclerView();
        addMystery();
    }

    private void setupRecyclerView() {
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        mysteryAdapter mAdapter = new mysteryAdapter(MainActivity.this, mysteries);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(recyclerView.getContext());
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.addItemDecoration(new DeviderItemDecoration(MainActivity.this,LinearLayoutManager.VERTICAL));
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(mAdapter);
    }

    void addMystery(){
        Mystery lunes=new Mystery("Lunes","Kalipay",R.drawable.joyful);
        mysteries.add(lunes);
        Mystery martes=new Mystery("Martes","Kasakit",R.drawable.joyful);
        mysteries.add(martes);
        Mystery merkules=new Mystery("Meyerkules","Himaya",R.drawable.joyful);
        mysteries.add(merkules);
        Mystery huwebes=new Mystery("Huwebes","Kahayag",R.drawable.joyful);
        mysteries.add(huwebes);
        Mystery beirnes=new Mystery("Byernes","Kasakit",R.drawable.joyful);
        mysteries.add(beirnes);
        Mystery sabado=new Mystery("Sabado","Kalipay",R.drawable.joyful);
        mysteries.add(sabado);
        Mystery domingo=new Mystery("Domingo","Himaya",R.drawable.joyful);
        mysteries.add(domingo);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/yamamz/santosrosaryo/MainActivity.kt
package com.yamamz.santosrosaryo

import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem

import com.yamamz.santosrosaryo.adapter.mysteryAdapter
import com.yamamz.santosrosaryo.model.Mystery
import com.yamamz.santosrosaryo.ui.DeviderItemDecoration
import com.yamamz.santosrosaryo.util.Util
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {

    private val mysteries = ArrayList<Mystery>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        setupRecyclerView()
        addMystery()


        fabEdit.setOnClickListener {
            val builder = AlertDialog.Builder(this, R.style.MyDialogTheme)
            val positiveText = getString(android.R.string.ok)
            builder.setTitle(getString(R.string.dialog_title))
            builder.setMessage(getString(R.string.dialog_message1))
            builder.setPositiveButton(positiveText) { _, _ ->
            }

            val dialog = builder.create()
            dialog.show()
        }

    }



    private fun setupRecyclerView() {


        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        val mAdapter = mysteryAdapter(this@MainActivity, mysteries)
        val mLayoutManager = LinearLayoutManager(recyclerView.context)
        recyclerView.layoutManager = mLayoutManager
        recyclerView.addItemDecoration(DeviderItemDecoration(this@MainActivity, LinearLayoutManager.VERTICAL))
        recyclerView.itemAnimator = DefaultItemAnimator()
        recyclerView.adapter = mAdapter
    }

    internal fun addMystery() {

        //joyFull
        val ramdomJoyful=Util.joyFull_array
        //sorrowful
        val ramdomsorrowful=Util.sorrowFull_array
        //gloriuos
        val ramdomGlorious=Util.glorious_array
        //luminous
        val ramdomluminous=Util.luminous_array
        val lunes = Mystery("Lunes", "Kalipay", ramdomJoyful[Random().nextInt(ramdomJoyful.size)])
        mysteries.add(lunes)
        val martes = Mystery("Martes", "Kasakit", ramdomsorrowful[Random().nextInt(ramdomGlorious.size)])
        mysteries.add(martes)
        val merkules = Mystery("Meyerkules", "Himaya", ramdomGlorious[Random().nextInt(ramdomGlorious.size)])
        mysteries.add(merkules)
        val huwebes = Mystery("Huwebes", "Kahayag", ramdomluminous[Random().nextInt(ramdomluminous.size)])
        mysteries.add(huwebes)
        val beirnes = Mystery("Byernes", "Kasakit", ramdomsorrowful[Random().nextInt(ramdomGlorious.size)])
        mysteries.add(beirnes)
        val sabado = Mystery("Sabado", "Kalipay", ramdomJoyful[Random().nextInt(ramdomJoyful.size)])
        mysteries.add(sabado)
        val domingo = Mystery("Domingo", "Himaya", ramdomGlorious[Random().nextInt(ramdomGlorious.size)])
        mysteries.add(domingo)
    }


}
